### DISCUSSION TITLE
Problem connecting to ControlLogix5583E

### DISCUSSION THREAD

**Discussion by AlexRegev** [ottowayi/pycomm3#185] [Q&A] [answered]
Hi,

I am trying to connect to a ControlLogix5583 using the CIPDriver.
when I try to list the identity i get the following info on my PLC:

`from pycomm3 import CIPDriver`
`print(CIPDriver.list_identity('192.168.1.2'))`

Output:
`{'encap_protocol_version': 1, 'ip_address': '192.168.1.2', 'vendor': 'Rockwell Automation/Allen-Bradley', 'product_type': 'Programmable Logic Controller', 'product_code': 166, 'revision': {'major': 33, 'minor': 11}, 'status': b'p1', 'serial': '00c8ffbd', 'product_name': '1756-L83E/B', 'state': 3}
`

When I try to open the connection I see the following information:
`driver = CIPDriver('192.168.1.2')`
`driver.open()`
`print(driver)`

Output:
`Device: None, Revision: -1.-1`

And if I try to read the module info I get this exception:
`driver = CIPDriver('192.168.1.2')`
`driver.open()`
`print(driver.get_module_info(0))`

Output:
`Traceback (most recent call last):`
`  File "C:\Users\AlexRegev\AppData\Local\Programs\Python\Python38\lib\site-packages\pycomm3\cip_driver.py", line 299, in get_module_info`
`    raise ResponseError(f"generic_message did not return valid data - {response.error}")`
`pycomm3.exceptions.ResponseError: generic_message did not return valid data - Connection failure (see extended status) - Unconnected message timeout  (01, 204)`

`The above exception was the direct cause of the following exception:`

`Traceback (most recent call last):`
`  File "C:/git/opcua-client-gui/test/simple_test.py", line 8, in <module>`
 `   print(driver.get_module_info(0))`
`  File "C:\Users\AlexRegev\AppData\Local\Programs\Python\Python38\lib\site-packages\pycomm3\cip_driver.py", line 302, in get_module_info`
 `   raise ResponseError("error getting module info") from err`
`pycomm3.exceptions.ResponseError: error getting module info`

Can you help me understand what could be the problem? am I missing some configuration?

Thanks a lot for your help.

Alex

**Comment by AlexRegev** [comment, marked as answer]
I found the issue, I wasn't addressing the PLC slot (located in slot 1).

that worked for me.

`with LogixDriver('192.168.1.2/1') as plc:`
    `print(plc.get_tag_list())`


Thanks

**Reply by ottowayi** [reply]
Glad you figured it out.  The `CIPDriver` is for non-PLC devices, the `LogixDriver` is the one you want.  It makes some assumptions about the device that is specific to PLCs and has the services you'd want for reading/writing tags and stuff.   Also, the `/1` is shorthand for `/backplane/1` and only the `LogixDriver` supports it.  You'd need to use the full path (`192.168.1.2/bp/1`) for it to work in the `CIPDriver`
